### PR TITLE
Dyna debug ip subdev

### PR DESCRIPTION
**Pull request by chienwei-lan**

### CODE REVIEW

File: src/runtime_src/core/include/experimental/xrt-next.h [hunk 1]
```diff
@@ -84,6 +84,11 @@ XCL_DRIVER_DLLESPEC
 int
 xclGetDebugProfileDeviceInfo(xclDeviceHandle handle, xclDebugProfileDeviceInfo* info);
 
+XCL_DRIVER_DLLESPEC
+int 
+xclGetSubdevPath(xclDeviceHandle handle,  const char* subdev,
+                 uint32_t idx, char* sysfsPath, size_t size);
+
```
Comment (stsoe) [inline src/runtime_src/core/include/experimental/xrt-next.h]: Please document.  
Also is this a public API?   This won't work on windows, I don't think it should be exposed in xrt-next.h

File: src/runtime_src/xrt/device/device.h [hunk 2]
```diff
@@ -869,6 +869,12 @@ class device : public xrt_device
     return m_hal->getSysfsPath(subdev, entry);
   }
 
+  hal::operations_result<std::string>
+  getSubdevPath(const std::string& subdev, uint32_t idx)
+  {
+    return m_hal->getSubdevPath(subdev, idx);
+  }
+
```
Comment (stsoe) [inline src/runtime_src/xrt/device/device.h]: No, this API should not be at this level.  It won't be implementable on windows.
It's fine to have the API in linux/shim.*, but not in common code.

File: src/runtime_src/xrt/device/hal.h [hunk 3]
```diff
@@ -661,6 +661,12 @@ class device
     return operations_result<std::string>();
   }
 
+  virtual operations_result<std::string>
+  getSubdevPath(const std::string& subdev, uint32_t idx)
+  {
+    return operations_result<std::string>();
+  }
```
Comment (stsoe) [inline src/runtime_src/xrt/device/hal.h]: Same comment as earlier

File: src/runtime_src/xrt/device/hal2.h [hunk 4]
```diff
@@ -652,6 +652,21 @@ class device : public xrt::hal::device
...
+    constexpr size_t max_path = 256;
+    char path_buf[max_path];
+    if (m_ops->mGetSubdevPath(m_handle, subdev.c_str(), idx, path_buf, max_path)) {
+      return hal::operations_result<std::string>();
+    }
+    path_buf[max_path - 1] = '\0';
+    std::string sysfs_path = std::string(path_buf);
+    return sysfs_path;
+  }
+
```
Comment (stsoe) [inline src/runtime_src/xrt/device/hal2.h]: Ditto.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/aim.c [hunk 5]
```diff
@@ -0,0 +1,228 @@
+/*
+ * A GEM style device manager for PCIe based OpenCL accelerators.
+ *
+ * Copyright (C) 2019 Xilinx, Inc. All rights reserved.
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/aim.c]: copyright yr is 2020

File: src/runtime_src/core/include/experimental/xrt-next.h (hunk 1 above)
Comment (chienwei-lan) [reply src/runtime_src/core/include/experimental/xrt-next.h]: removed, I found some other APIs such as 

xclCreateProfileResults(xclDeviceHandle, struct ProfileResults**);
xclGetProfileResults(xclDeviceHandle, struct ProfileResults*);
xclDestroyProfileResults(xclDeviceHandle, struct ProfileResults*);

it's only for XDP internal use, should we remove them from xrt-next.h?

File: src/runtime_src/xrt/device/hal2.h (hunk 4 above)
Comment (chienwei-lan) [reply src/runtime_src/xrt/device/hal2.h]: change "sysfs_path" to more general one "path"

**Pull request merged**